<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from ./description/environment.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="environment">

  <!-- Common -->
  <link name="env_base"/>
  <!-- Table -->
  <link name="table_link" type="fixed">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="./environment/collision/table.stl"/>
      </geometry>
      
    </collision>
    <inertial>
            <mass value="0"/>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <inertia ixx="0" ixy="0.0" ixz="0.0" iyy="0" iyz="0" izz="0"/>
        </inertial>
  </link>
  <joint name="env_table_joint" type="fixed">
    <parent link="env_base"/>
    <child link="table_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.36 -0.37"/>
  </joint>
  <!-- 3D Printer -->
  <link name="3d_printer_link" type="fixed">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="./environment/collision/3d_printer.stl"/>
      </geometry>
      
    </collision>
    <inertial>
            <mass value="0"/>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <inertia ixx="0" ixy="0.0" ixz="0.0" iyy="0" iyz="0" izz="0"/>
        </inertial>
  </link>
  <joint name="env_3d_printer_joint" type="fixed">
    <parent link="env_base"/>
    <child link="3d_printer_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="-0.28 0.32 0.3"/>
  </joint>
  <!-- Blade Conveyor -->
  <link name="blade_conveyor_link" type="fixed">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="./environment/collision/conveyor.stl"/>
      </geometry>
      
    </collision>
    <inertial>
            <mass value="0"/>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <inertia ixx="0" ixy="0.0" ixz="0.0" iyy="0" iyz="0" izz="0"/>
        </inertial>
  </link>
  <joint name="env_blade_conveyor_joint" type="fixed">
    <parent link="env_base"/>
    <child link="blade_conveyor_link"/>
    <origin rpy="0.0 0.0 1.5707963" xyz="-0.85 -0.25 -0.75"/>
  </joint>
  <!-- Blade Feeder -->
  <link name="blade_feeder_link" type="fixed">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="./environment/collision/blade_feeder.stl"/>
      </geometry>
      
    </collision>
    <inertial>
            <mass value="0"/>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <inertia ixx="0" ixy="0.0" ixz="0.0" iyy="0" iyz="0" izz="0"/>
        </inertial>
  </link>
  <joint name="env_blade_feeder_joint" type="fixed">
    <parent link="env_base"/>
    <child link="blade_feeder_link"/>
    <origin rpy="0.0 0.0 1.5707963" xyz="-0.85 -0.25 -0.75"/>
  </joint>
  <!-- Assembly Jig -->
  <link name="assembly_jig_link" type="fixed">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="./environment/collision/assembly_jig.stl"/>
      </geometry>
      
    </collision>
    <inertial>
            <mass value="0"/>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <inertia ixx="0" ixy="0.0" ixz="0.0" iyy="0" iyz="0" izz="0"/>
        </inertial>
  </link>
  <joint name="env_assembly_jig_joint" type="fixed">
    <parent link="env_base"/>
    <child link="assembly_jig_link"/>
    <origin rpy="0.0 1.5707963 1.5707963" xyz="0.2 0.28 0.14"/>
  </joint>
  <!-- Knife Feeder -->
  <link name="knife_feeder_link" type="fixed">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="./environment/collision/knife_feeder.stl"/>
      </geometry>
      
    </collision>
    <inertial>
            <mass value="0"/>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <inertia ixx="0" ixy="0.0" ixz="0.0" iyy="0" iyz="0" izz="0"/>
        </inertial>
  </link>
  <joint name="env_knife_feeder_joint" type="fixed">
    <parent link="env_base"/>
    <child link="knife_feeder_link"/>
    <origin rpy="0.0 0.0 -1.5707963" xyz="0.85 -0.25 -0.75"/>
  </joint>
  <!-- Knife Conveyor -->
  <link name="knife_conveyor_link" type="fixed">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="./environment/collision/conveyor.stl"/>
      </geometry>
      
    </collision>
    <inertial>
            <mass value="0"/>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <inertia ixx="0" ixy="0.0" ixz="0.0" iyy="0" iyz="0" izz="0"/>
        </inertial>
  </link>
  <joint name="env_knife_conveyor_joint" type="fixed">
    <parent link="env_base"/>
    <child link="knife_conveyor_link"/>
    <origin rpy="0.0 0.0 -1.5707963" xyz="0.85 -0.25 -0.75"/>
  </joint>
</robot>
